src/soc/ibm/power9/chip.c

```diff
@@ -1,6 +1,75 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+
 #include <device/device.h>
+#include <fit.h>
+#include <cpu/power/istep_13.h>
+#include "istep_13_scom.h"
+
+static int dt_platform_fixup(struct device_tree_fixup *fixup,
+			      struct device_tree *tree)
+{
+	struct device_tree_node *node;
+
+	/* Memory devices are always direct children of root */
+	list_for_each(node, tree->root->children, list_node) {
+		const char *devtype = dt_find_string_prop(node, "device_type");
+		if (devtype && !strcmp(devtype, "memory")) {
+			dt_add_u32_prop(node, "ibm,chip-id", 0 /* FIXME for second CPU */);
+		}
+	}
+
+	return 0;
+}
+
+#define SIZE_MASK	PPC_BITMASK(13,23)
+#define SIZE_SHIFT	(63 - 23)
+#define BASE_MASK	PPC_BITMASK(24,47)
+#define BASE_SHIFT	(63 - 47)
+
+/* Value in registers is in 4GB units, ram_resource() expects kilobytes. */
+#define TO_KB(x)	((x) << 22)
```

Comment:
Looking at the name of the macro alone, one may be unsure, what is converted to kilobytes.
Please consider renaming this macro. I would suggest something like `CONVERT_4GB_TO_KB`

Reply:
Done